blocks/bricks/sequence_generators.py

```diff
@@ -436,6 +436,9 @@ def emit(self, readouts):
 
     @application
     def cost(self, readouts, outputs):
+        # WARNING: unfortunately this application method works
+        # just fine when `readouts` and `outputs` have
+        # different dimensions. Be careful!
         probs = self.probs(readouts)
```

Comment:
I didn't know about the `Assert` op and it kind of looks like what I had in mind. I don't think I'll have much time to look into this in detail now but I'll think about it. I was mainly curious to know whether I was missing out on some well-known feature. Thanks for the info!